### PR TITLE
Use a find-module to locate Dawn/WebGPU

### PR DESCRIPTION
**Pull request by alexreinking**
Might want to add some documentation in `README_webgpu.md` for how AOT compilation works.

### CODE REVIEW

**Review by steven-johnson — APPROVED** [review]

File: cmake/HalideGeneratorHelpers.cmake [hunk 1]
```diff
@@ -718,9 +718,8 @@ function(_Halide_target_link_gpu_libs TARGET VISIBILITY)
     endif ()
 
     if ("${ARGN}" MATCHES "webgpu")
-        if (DEFINED ENV{HL_WEBGPU_NATIVE_LIB})
-            target_link_libraries(${TARGET} INTERFACE $ENV{HL_WEBGPU_NATIVE_LIB})
-        endif ()
+        find_package(Halide_WebGPU REQUIRED)
+        target_link_libraries(${TARGET} ${VISIBILITY} Halide::WebGPU)
```
[the remark this answers is not in the document]
Comment (steven-johnson) [reply cmake/HalideGeneratorHelpers.cmake:722]: Tagging @jrprice as an FYI.

**Comment by steven-johnson** [comment]
...why aren't the buildbots running on this?

**Comment by alexreinking** [comment]
> ...why aren't the buildbots running on this?

Maybe because it's a pr-to-pr?

**Comment by steven-johnson** [comment]
> > ...why aren't the buildbots running on this?
> 
> Maybe because it's a pr-to-pr?

ahhh, gotcha, let's land it then I guess

**Pull request merged**